### PR TITLE
Add ability to use default claim dialect for OIDC authenticators

### PR DESCRIPTION
**Pull request by janakamarasena**
Partially solves https://github.com/wso2/product-is/issues/5437

Gives the ability for a user to specify if the OIDC claim dialect should be used for defined claim mappings from the IDP UI for OIDC based authenticator. By default, the checkbox is checked to preserve the old behaviour.

Screen Shot 2019-06-17 at 1 32 23 PM

This PR only adds the UI capability and methods needed for the actually functionality to be implemented.

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/ApplicationAuthenticator.java [hunk 1]
```diff
@@ -85,6 +85,18 @@ public AuthenticatorFlowStatus process(HttpServletRequest request,
      */
     public String getClaimDialectURI();
 
+    /**
+     * Get the claim dialect URI if this authenticator receives claims in a standard dialect
+     * and needs to be mapped to the Carbon dialect http://wso2.org/claims
```
Comment (Abilashini) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/ApplicationAuthenticator.java]: Java Doc description should end with a period (.)

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java [hunk 2]
```diff
@@ -1311,6 +1311,15 @@ private static void buildOpenIDConnectAuthenticationConfiguration(IdentityProvid
         }
         properties[8] = property;
 
+        property = new Property();
+        property.setName(IdentityApplicationConstants.Authenticator.OIDC.USE_OIDC_CLAIM_DIALECT);
+        if ("on".equals(paramMap.get(IdentityApplicationConstants.Authenticator.OIDC.USE_OIDC_CLAIM_DIALECT))) {
```
Comment (Abilashini) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java]: Better to have a constant for "on"

Comment (Abilashini) [reply components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java]: Can't we directly set the condition as the value to the property? Hope if-else condition is not necessary here

Comment (janakamarasena) [reply components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java]: Added as a constant. This is a checkbox and HTML forms don't submit a property for checkboxes when they are not checked. That's why we need the if-else condition.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/ApplicationAuthenticator.java (hunk 1 above)
Comment (janakamarasena) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/ApplicationAuthenticator.java]: fixed.

**Comment by janakamarasena** [comment]
This PR was reverted.

**Pull request merged**